stravalib/attributes.py

```diff
@@ -229,7 +234,11 @@ def unmarshal(self, v):
         more complex types, where subclasses will override this behavior.
         """
         if not isinstance(v, timedelta):
-            v = timedelta(seconds=v)
+            if isinstance(v, str) or isinstance(v, unicode):
+                h,m,s = v.split(':')
+                v = timedelta(seconds = int(h)*3600 + int(m)*60 + int(s))
+            else:
+                v = timedelta(seconds=v)
         return v
 
     def marshal(self, v):
```

Comment:
I like that. Was thinking about it too.

One question about your patch - shouldn't it remove the original `return str(v) if v else None` line?

Reply:
Yes, probably, but I am not sure if we always have a `timedelta` object to start with, plus we should still handle a value of `None`. It seemed more resilient to keep it, especially since your `unmarshall` method handles strings. Totally your call or eventually @hozn I guess.

Reply:
I agree that leaving it resilient makes sense!